pyro/infer/mcmc/nuts.py

```diff
@@ -6,11 +6,20 @@
 
 import pyro
 import pyro.distributions as dist
-from pyro.distributions.util import logsumexp, scalar_like
+from pyro.distributions.util import scalar_like
 from pyro.infer.mcmc.hmc import HMC
 from pyro.ops.integrator import velocity_verlet
 from pyro.util import optional, torch_isnan
 
+
+def _logsumexp(vals, dim=0):
```

Comment:
Would suggest simplifying this to take in the values directly so that we can avoid `stack` below. Also, since we know that the values are scalars, we can remove the dim arg and the asserts as this is only internally used.

```python
def _logaddexp(x, y):
    minval, maxval = (x, y) if x < y else (y, x)
    return (maxval - minval).exp().log() + maxval    
```

Reply:
^ should be log1p above; corrected.

Reply:
Yeah, I think this version is better. Should it be min - max instead of max - min?

Reply:
```python
def _logaddexp(x, y):
    minval, maxval = (x, y) if x < y else (y, x)
    return (minval - maxval).exp().log1p() + maxval
```

Reply:
You're right, should be min - max.